src/lib/session.ts

```diff
@@ -1,40 +1,30 @@
 import { derived, get, writable } from "svelte/store";
 
-export interface Session {
-  id: string;
-  publicKey: string;
-}
+import { HttpdClient } from "@httpd-client";
 
-interface SessionResponse {
-  sessionId: string;
-  status: string;
+export interface StoredSession {
+  id: string;
   publicKey: string;
-  issuedAt: number;
-  expiresAt: number;
 }
 
-const store = writable<Session | undefined>(undefined);
+const store = writable<StoredSession | undefined>(undefined);
 export const sessionStore = derived(store, s => s);
 
-const endpoint = "http://localhost:8080/api/v1/sessions";
+const api = new HttpdClient("http://localhost:8080");
 
 export async function authenticate(params: {
   id: string;
   signature: string;
   publicKey: string;
-}): Promise<"success" | "failure"> {
+}): Promise<boolean> {
   disconnect();
 
-  const request = await fetch(`${endpoint}/${params.id}`, {
-    method: "PUT",
-    headers: { "Content-Type": "application/json" },
-    body: JSON.stringify({ sig: params.signature, pk: params.publicKey }),
-  });
-  if (request.ok) {
+  try {
+    await api.session.create(params.id, params.signature, params.publicKey);
     save(params.id, params.publicKey);
-    return "success";
-  } else {
-    return "failure";
+    return true;
+  } catch {
+    return false;
```

Comment:
This is probably an intentional shortcut but I want to highlight that you’re not disambiguating errors here. For example, if you try to authenticate but the server is not reachable you return `false`. I don’t yet know how this is used but this may lead to situations where you tell the user that their credentials are not valid when in fact the server has crashed.

Reply:
Yeah, this is a shortcut for now. We'll improve on this in a follow up.